checks.d/consul.py

```diff
@@ -118,24 +151,30 @@ def _check_for_leader_change(self, instance):
             self._last_known_leader = leader
             return
 
+        agent = self._get_agent_url(instance)
+        agent_dc = self._get_agent_datacenter(instance)
+
         if leader != self._last_known_leader:
-            self.log.info(('Leader change from {0} to {1}. Sending new leader event').format(
-                self._last_known_leader, leader))
-
-            self.event({
-                "timestamp": int(datetime.now().strftime("%s")),
-                "event_type": "consul.new_leader",
-                "source_type_name": self.SOURCE_TYPE_NAME,
-                "msg_title": "New Consul Leader Elected in consul_datacenter:{0}".format(agent_dc),
-                "aggregation_key": "consul.new_leader",
-                "msg_text": "The Node at {0} is the new leader of the consul datacenter {1}".format(
-                    leader,
-                    agent_dc
-                ),
-                "tags": ["prev_consul_leader:{0}".format(self._last_known_leader),
-                         "curr_consul_leader:{0}".format(leader),
-                         "consul_datacenter:{0}".format(agent_dc)]
-            })
+            # There was a leadership change
+            if perform_new_leader_checks or (perform_self_leader_check and agent == leader):
+                # We either emit all leadership changes or emit when we become the leader and that just happened
+                self.log.info(('Leader change from {0} to {1}. Sending new leader event').format(
+                    self._last_known_leader, leader))
+
+                self.event({
+                    "timestamp": int(datetime.now().strftime("%s")),
+                    "event_type": "consul.new_leader",
```

Comment:
Again, not your fault at all, but `consul.new_leader` should probably go in a constant. It's not your job to cleanup this, so feel free to ignore.